plugin/ochttp/propagation/tracecontext/propagation.go

```diff
@@ -98,13 +104,26 @@ func (f *HTTPFormat) SpanContextFromRequest(req *http.Request) (sc trace.SpanCon
 	return sc, true
 }
 
+func getRequestHeader(req *http.Request, name string, commaSeparated bool) (hdr string, ok bool) {
+	ok = false
+	for k, v := range req.Header {
+		if strings.EqualFold(k, name) {
```

Comment:
Instead of looping through all the headers, you can canonicalize `name` using `textproto.CanonicalMIMEHeaderKey` and index `req.Header` directly.

Reply:
Is this conventional (which might break later) or something we could rely on?

Reply:
It is documented here: https://golang.org/pkg/net/http/?#Header.Get

Reply:
That was what I was looking at, and it's not working.

According to the doc:

Get gets the **first** value associated with the given key.
...
To **access multiple values of a key**, or to use non-canonical keys, **access the map directly**.

Reply:
Thanks for the explanation, makes sense.